### PR TITLE
Fixed #30304 -- Added support for HttpOnly, SameSite, and Secure flags on language cookies.

### PR DESCRIPTION
**Pull request by bluetech**
https://code.djangoproject.com/ticket/30304

### CODE REVIEW

**Comment by bluetech** [comment]
The CI failure is unrelated:

```
hudson.plugins.git.GitException: Failed to fetch from git://github.com/django/django.git
```

File: docs/releases/3.0.txt [hunk 1]
```diff
@@ -153,7 +153,11 @@ Generic Views
 Internationalization
 ~~~~~~~~~~~~~~~~~~~~
 
-* ...
+* Added the :setting:`LANGUAGE_COOKIE_HTTPONLY`,
+  :setting:`LANGUAGE_COOKIE_SECURE` and :setting:`LANGUAGE_COOKIE_SAMESITE`
+  settings to set the ``HttpOnly``, ``Secure`` and ``SameSite`` flags on the
+  language cookie. The default values of these settings maintain the previous
+  behavior.
```
Comment (felixxm) [inline docs/releases/3.0.txt]:
```
* Added the :setting:`LANGUAGE_COOKIE_HTTPONLY`,
  :setting:`LANGUAGE_COOKIE_SAMESITE`, and :setting:`LANGUAGE_COOKIE_SECURE` 
  settings to set the ``HttpOnly``, ``SameSite``, and ``Secure`` flags on
  language cookies. The default values of these settings preserve the previous
  behavior.
```

**Comment by bluetech** [comment]
Thanks for the review @felixxm, I applied your comments.

**Comment by matthiask** [comment]
Re. system checks for language cookies. Even on sites only using HTTPS I probably wouldn't want to set the language cookie's secure flag because e.g. when redirecting from HTTP to HTTPS and maybe also redirecting from `/` to `/en/` or `/es/`, maybe using `i18n_patterns`. I would not want to fall back to the browser's `accept-language` header when a language cookie has already been set earlier.

Or maybe I'm overseeing security issues related to the language cookie? I've read the ticket description but I don't see the value in pushing developers towards setting those values (yet?)

**Comment by carltongibson** [comment]
Hey @matthiask. Yes, quite possibly. I'm never quite sure to what level we want "everything HTTPS" these days. (I imagine it's more privacy than security per se in a lot of cases.)

**Comment by bluetech** [comment]
I agree with @matthiask. I think the default values for these flags are suitable, so there is no reason to nudge people to set them.

**Comment by adamchainz** [comment]
I'd rather have the system check to push people to turn them on. They can silence the check if it's not suitable for their use case.

I don't quite follow your double-redirect concern @matthiask , since `SecurityMiddleware` will redirect to HTTPS before a language redirect would be done?

**Pull request merged**